### ISSUE TITLE
Option for excluding individual from its group when the individual is compared with the group

### ISSUE THREAD

**Issue by martynamolak** [poseidon-framework/poseidon-analysis-hs#32]
There are situations when one wants to calculate f-statistics between an individual and its own population, e.g.:
F4(PopulationA, TestPopulation; Individual1FromPopulationA, Outgroup)

In such situations it might be useful (or might make more sense) to use PopulationA but with Individual1 excluded.
I'm not sure whether this should be a default behavior but would certainly be a useful option.
Specifying an ad hoc "PopulationA-minus-Individual1FromPopulationA" for each individual is of course an option but when one is estimating a lot of f-statistics and, particularly, when there is many individuals that one would like such estimates for, it ends up being fairly cumbersome.

If this would be a fairly easy option to introduce, it would be awesome to have it.

**Comment by stschiff** [comment]
Right. Thanks for adding this issue. I agree about the usefulness, but so far lack imagination for how the exact API could look so that users won't be surprised. Will think about it.

**Comment by martynamolak** [comment]
Don't you think that it actually could be a default behavior that whenever an individual is compared to its own population it gets excluded from the population? Then it wouldn't be confusing by there being more options to set up.

I mean every time you have fstats estimation, xerxes would first check whether the entities which are individuals are a part of any other entity that are populations or packages and if they are, turn this population/package into (population - individual) or (package - individual).
That should also work in cases when there are more than one individual in the fstats.
Eg.

fstats:
- type: F4
  a: PopA
  b: PopB
  c: PopAInd1
  d: PopBInd1

would turn be automatically processed into:
fstats:
- type: F4
  a: PopA minus PopAInd1
  b: PopB minus PopBInd1
  c: PopAInd1
  d: PopBInd1

and:

fstats:
- type: F4
  a: PopA
  b: PopB
  c: PopAInd1
  d: PopAInd2

would turn be automatically processed into:
fstats:
- type: F4
  a: PopA minus PopAInd1 minus PopAInd2
  b: PopB minus PopBInd1
  c: PopAInd1
  d: PopAInd2

**Comment by stschiff** [comment]
Hmm, yes, I admit that sounds like a useful feature.
